bundles/org.openhab.binding.http/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,158 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="http"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="url"
+		extensible="color,contact,datetime,dimmer,image,location,number,rollershutter, string,switch">
+		<label>HTTP URL Thing</label>
+		<description>Represents a base URL and all associated requests.</description>
+
+		<config-description>
+			<parameter name="baseURL" type="text" required="true">
+				<label>Base URL</label>
+				<description>The URL set here can be extended in the channel configuration.</description>
+				<context>network-address</context>
+			</parameter>
+			<parameter name="refresh" type="integer" unit="s" min="0">
+				<label>Refresh Time</label>
+				<description>Time between two refreshes of all channels</description>
+				<default>30</default>
+			</parameter>
+			<parameter name="timeout" type="integer" unit="ms" min="0">
+				<label>Timeout</label>
+				<description>The timeout in ms for each request</description>
+				<default>3000</default>
+			</parameter>
+			<parameter name="username" type="text">
+				<label>Username</label>
+				<description>Basic Authentication username</description>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="password" type="text">
+				<label>Password</label>
+				<description>Basic Authentication password</description>
+				<context>password</context>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="authMode" type="text">
+				<label>Authentication Mode</label>
+				<options>
+					<option value="BASIC">Basic Authentication</option>
+					<option value="DIGEST">Digest Authentication</option>
+				</options>
+				<default>BASIC</default>
+				<limitToOptions>true</limitToOptions>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="commandMethod" type="text">
+				<label>Command Method</label>
+				<description>HTTP method (GET,POST, PUT) for sending commands.</description>
+				<options>
+					<option value="GET">GET</option>
+					<option value="POST">POST</option>
+					<option value="PUT">PUT</option>
+				</options>
+				<limitToOptions>true</limitToOptions>
+				<default>GET</default>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="contentType" type="text">
+				<label>Content Type</label>
+				<description>The MIME content type. Only used for `POST` and `PUT`.</description>
+				<options>
+					<option value="application/json">application/json</option>
+					<option value="application/xml">application/xml</option>
+					<option value="text/html">text/html</option>
+					<option value="text/plain">text/plain</option>
+					<option value="text/xml">text/xml</option>
```

Comment:
This list is hardly exhaustive, should you allow users to input it manually instead?